docs/user-guide/best-practices/udf-best-practices.mdx

```diff
@@ -0,0 +1,311 @@
+
+## A short reminder: The anatomy of a UDF
+
+```python showLineNumbers
+@fused.udf
+def udf(my_awesome_input: int = 1):
+    import pandas as pd
+
+    return pd.DataFrame({"Look at my output: ": [my_awesome_input]})
+```
+
+Each UDF has a few specific elements:
+- The [`@fused.udf` decorator](/core-concepts/write/#fusedudf-decorator)
+- Arguments -ideally typed-
+- imports _inside_ the function
+- Some logic
+- A [supported `return` object](/core-concepts/write/#return-object)
+
+:::note
+All of this is explained in the ["Write UDF"](/core-concepts/write/) section in much more details
+:::
+
+You can then run UDFs from _anywhere_ you have access to Python with [`fused.run(udf)`](/core-concepts/run-udfs/run-small-udfs/#fusedrun). These are still Python functions, giving you a lot of flexibility on what oyu can do, but we have some recommendations for keeping them fast & efficient.
+
+## Writing efficient UDFs
+
+### Keep things small
+
+The main benefit of Fused UDFs is how responsive they are. To do so, they run on Python serverless compute, but [quickly timeout](/core-concepts/run-udfs/run-small-udfs/#defining-small-job). The best way to do that is to keep **things** fast is to keep them small:
+
+- Break pipelines into single tasks UDFs
+- Leverage [`fused.run()`](/core-concepts/run-udfs/run-small-udfs/#fusedrun) to chain together UDFs
+
+<details>
+    <summary>Example: Breaking down a complex pipeline into smaller UDFs</summary>
+
+    ❌ Not recommended:
+
+    ```python showLineNumbers
+    @fused.udf
+    def inefficient_pipeline_udf(data_path):
+        import pandas as pd
+
+        df = pd.read_csv(data_path)
+        # Some complicated processing logic to create df_processed
+        processed_df = ... 
+
+        return processed_df
+    ```
+
+    ✅ Instead, break it down:
+
+    ```python showLineNumbers
+    @fused.udf
+    def load_data_udf(data_path):
+        import pandas as pd
+        return pd.read_csv(data_path)
+    ```
+
+    ```python showLineNumbers
+    @fused.udf
+    def process_data_udf(df):
+        import pandas as pd
+
+        # Some complicated processing logic to create df_processed
+        processed_df = ... 
+
+        return processed_df
+    ```
+
+    ```python showLineNumbers
+    @fused.udf
+    def pipeline_udf(data_path):
+        import pandas as pd
+
+        df = fused.run(load_data_udf, data_path=data_path)
+        processed_df = fused.run(process_data_udf, df=df)
+
+        return processed_df
+    ```
+</details>
+
+### Run often, Iterate quickly
+
+Just like writing short cells when developing in a Jupyter Notebook, we recommend you keep your UDFs short & fast to execute
+
+⚡️ Aim for **UDFs that take 30s - 1min to run**
+
+UDFs run with [`fused.run()`](/core-concepts/run-udfs/run-small-udfs/#fusedrun) [timeout after 120s](/core-concepts/run-udfs/run-small-udfs/#defining-small-job) so we recommend you keep a buffer in case your UDF takes a bit longer to execute
```

Comment:
```suggestion
UDFs run with [`fused.run()`](/core-concepts/run-udfs/run-small-udfs/#fusedrun) [time out after 120s](/core-concepts/run-udfs/run-small-udfs/#defining-small-job) so we recommend you keep a buffer in case your UDF takes a bit longer to execute
```